src/rest/SearchEndpoint.ts

```diff
@@ -1434,12 +1434,29 @@ function includeAnalytics() {
   return function (target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
     const { originalMethod, nbParams } = decoratorSetup(target, key, descriptor);
     descriptor.value = function (this: SearchEndpoint, ...args: any[]) {
-      const analyticsInstance = new AnalyticsInformation();
+      const analyticsInstance = (args && args[1] && (args[1].analyticsInformation as AnalyticsInformation)) || new AnalyticsInformation();
+      const actionCause = analyticsInstance.pendingSearchEvent ? analyticsInstance.pendingSearchEvent.getEventCause() : null;
+      const customData = analyticsInstance.pendingSearchEvent ? analyticsInstance.pendingSearchEvent.getEventMeta() : null;
```

Comment:
I would slightly prefer if we made `pendingSearchEvent` on `AnalyticsInformation` a setter and exposed `pendingSearchEventCause`/`pendingSearchEventMeta` as getters instead.